scripts/check_names.py

```diff
@@ -727,7 +623,141 @@ def parse_identifiers(self, include, exclude=None):
 
     def parse_symbols(self):
         """
-        Compile the Mbed TLS libraries, and parse the TLS, Crypto, and x509
+        Compile a library, and parse the object files using nm to retrieve the
+        list of referenced symbols. Exceptions thrown here are rethrown because
+        they would be critical errors that void several tests, and thus needs
+        to halt the program. This is explicitly done for clarity.
+
+        Returns a List of unique symbols defined and used in the libraries.
+        """
+        raise NotImplementedError("parse_symbols must be implemented by a code parser")
+
+    def comprehensive_parse(self):
+        """
+        (Must be defined as a class method)
+        Comprehensive ("default") function to call each parsing function and
+        retrieve various elements of the code, together with the source location.
+
+        Returns a dict of parsed item key to the corresponding List of Matches.
+        """
+        raise NotImplementedError("comprehension_parse must be implemented by a code parser")
```

Comment:
What about to just remove them from CodeParser?

Reply:
I think these should stay to retain compatibilty. These are abstract methods that all code parsers, present and future must implement to be used interchangabily.

This should hopefully enforce a more rigid code parser test script.